### ISSUE TITLE
FastGrowCut is duplicated in the CarreraSlice extension

### ISSUE THREAD

**Issue by jcfr** [ljzhu/FastGrowCut#5]
Should the version in CarreraSlice be removed ? 
See https://github.com/ikolesov/CarreraSlice/tree/master/src/fastGrowCut

Cc: @pieper @rkikinis @fedorov @lassoan @hjmjohnson

**Comment by pieper** [comment]
We need to ask Peter and Ivan if they are still available to maintain it.

**Comment by fedorov** [comment]
I didn't even know there are two versions.

**Comment by ljzhu** [comment]
Yes, the CarreraSlice has a copy of the fast growcut. It combines fast
growcut with other functions.

**Comment by jcfr** [comment]
@ljzhu Would it make sense to remove the duplicated code from CarreraSlice and depend on the existing FastGrowCut extension ?

**Comment by ljzhu** [comment]
Yes, it sounds good. A simple question: how to resolve the dependency
during the compiling process? The CarreraSlice uses the FastGrowCut as a
VTK object. Thanks!! --Liangjia

**Comment by jcfr** [comment]
As detailed in the developer FAQ entry (See [Can an extension depend on other extensions ?](http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/FAQ#Can_an_extension_depend_on_other_extensions_.3F)), if you specify that `CarreraSlice` depends on `FastGrowCut`, it should be possible to use `FastGrowCut` libraries within  `CarreraSlice`.  

The generated description file that will contributed to the [Slicer/ExtensionsIndex](https://github.com/Slicer/ExtensionsIndex) will contains the `depends` entry. For example, see [GelDosimetryAnalysis.s4ext#L15](https://github.com/Slicer/ExtensionsIndex/blob/3caf4f360ce0d413572a1e8a111d265f1c829ec6/GelDosimetryAnalysis.s4ext#L15))

On the Slicer factories,  `CarreraSlice` will then be build after  `FastGrowCut` and will be configured with `-DFastGrowCut_DIR:PATH=/path/to/FastGrowCut-build`

The only requirement for `FastGrowCut` is to provide a `FastGrowCutConfig.cmake`. There is currently no automatic way of generating this file. I could help for that part. The associated Slicer issue is [#2644](http://www.na-mic.org/Bug/view.php?id=2644)

**Comment by ljzhu** [comment]
Hi JC,

It would be great if you could help with generating the
FastGrowCutConfig.cmake file.

Thanks,
Liangjia
